### PR TITLE
Allow queueing dictionary update & delete together

### LINKED ISSUE #1557: Dictionary update queue
Currently we have to wait for a dictionary to be updated before update another one. 

Should be able to queue dictionary updates. Better if there is an option to update all dictionaries.

Similar to #1412

### PR DESCRIPTION
**Pull request by khaitruong922**
Closes #1557
~~When enqueue a dictionary, hide its update button.~~

~~Also refactor dictionary update and delete behavior, only allow 1 queue to run at a time.~~
~~- If update queue is running then disable all delete buttons.~~
~~- If delete queue is running then hide all update buttons.~~

### CODE REVIEW

File: ext/js/pages/settings/dictionary-controller.js [hunk 1]
```diff
@@ -1156,7 +1228,12 @@ export class DictionaryController {
         }
 
         await this._deleteDictionary(dictionaryTitle);
-        this._settingsController.trigger('importDictionaryFromUrl', {url: downloadUrl, profilesDictionarySettings});
+        /** @type {Promise<void>} */
+        const importPromise = new Promise((resolve) => {
+            this._settingsController.trigger('importDictionaryFromUrl', {url: downloadUrl, profilesDictionarySettings, onImportDone: resolve});
+        })
```
Comment (khaitruong922) [inline ext/js/pages/settings/dictionary-controller.js:1234]: I don't think this is the correct way to do this but this is the only way I can think of to await an event trigger.

**Comment by khaitruong922** [comment]
Ready for review. 
Now allow queueing dictionary deletes and updates in the same task queue.

**Review by jamesmaa — APPROVED** [review]
Didn't test this but code looks sensible

**Comment by khaitruong922** [comment]
Now will persist the update button with the same link even if the dictionary list is rerendered.

**Review by Kuuuube — APPROVED** [review]

**Pull request merged**